hubble/src/chain_id_query.rs

```diff
@@ -205,10 +219,9 @@ pub async fn tx(db: PgPool, indexers: Indexers) {
                 }
             }
             IndexerConfig::Eth(eth_config) => {
-                let provider =
-                    Provider::<Http>::try_from(eth_config.urls[0].clone().as_str()).unwrap();
+                let provider = ProviderBuilder::new().on_http(eth_config.urls[0].clone());
```

Comment:
<div>

<strong>Bito Code Review Agent Run</strong> <a href=https://github.com/unionlabs/union/pull/2693#issuecomment-2293655644>#ef6b49</a><span> - 08/16/2024, 03:05 pm</span>
<br>

<div><small>🔴</small> High importance</div>

<div id="suggestion">
<div id="issue">
<b>Issue:</b> The use of '.unwrap()' for error handling can cause the program to panic if the provider creation or chain ID retrieval fails. This is not ideal for a production environment. Similar issues were also found in:
- hubble/src/arb.rs (line 125)
- hubble/src/eth/indexer.rs (line 508-509)
</div>
<div id="fix">
<b>Fix:</b> Replace '.unwrap()' with proper error handling to ensure the program does not panic in case of an error.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -222,7 +222,7 @@
  let provider = ProviderBuilder::new().on_http(eth_config.urls[0].clone()).map_err(|e| {
      tracing::error!("Failed to create provider: {:?}", e);
      e
  })?;
 @@ -224,7 +224,7 @@
  let chain_id = provider.get_chain_id().await.map_err(|e| {
      tracing::error!("Failed to get chain ID: {:?}", e);
      e
  })?.to_string();
```
</div>
</details>
</div>
<br>


</div>

Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged